titus-common/src/main/java/com/netflix/titus/common/util/CollectionsExt.java

```diff
@@ -223,6 +223,12 @@ private CollectionsExt() {
         return result;
     }
 
+    public static <K, V> Map<K, V> copyAndRemoveByKey(Map<K, V> original, Predicate<K> keysPredicate) {
+        return original.entrySet().stream()
```

Comment:
Stream implementation has poor performance, which may matter in some cases. Direct for loop would do much better here.

Reply:
I agree there's a performance hit, but have you seen it being the actual bottleneck in practice for the control plane? We already use the stream api on the similar `copyAndRemoveByValue`, and since this is only being used during job submissions, I'd say avoid the early optimization for now.

Reply:
Yes, we have seen this in Fenzo. As a general rule we avoided streams API in all functions called by Fenzo.